includes/class-wc-ajax.php

```diff
@@ -1650,16 +1736,16 @@ public static function refund_line_items() {
 			$max_refund  = wc_format_decimal( $order->get_total() - $order->get_total_refunded(), wc_get_price_decimals() );
 
 			if ( ! $refund_amount || $max_refund < $refund_amount || 0 > $refund_amount ) {
-				throw new exception( __( 'Invalid refund amount', 'woocommerce' ) );
+				throw new Exception( __( 'Invalid refund amount', 'woocommerce' ) );
 			}
 
-			if ( $refunded_amount !== wc_format_decimal( $order->get_total_refunded(), wc_get_price_decimals() ) ) {
-				throw new exception( __( 'Error processing refund. Please try again.', 'woocommerce' ) );
+			if ( wc_format_decimal( $order->get_total_refunded(), wc_get_price_decimals() ) !== $refunded_amount ) {
+				throw new Exception( __( 'Error processing refund. Please try again.', 'woocommerce' ) );
 			}
 
 			// Prepare line items which we are refunding.
 			$line_items = array();
-			$item_ids   = array_unique( array_merge( array_keys( $line_item_qtys, $line_item_totals ) ) );
+			$item_ids   = array_unique( array_merge( array_keys( $line_item_qtys, $line_item_totals, true ) ) );
```

Comment:
Not sure if misread but this line is grabbing all possible item IDs from the post data, so each ID can be handled later on in tge method. The values are irrelevent.

Reply:
...but it will never return any IDs.

`array_keys( $line_item_qtys, $line_item_totals, true )` means it will return keys from `$line_item_qtys` where value is strictly equal to an array `$line_item_totals`. That will never happen.

Reply:
This is resolved now @peterfabian